### ISSUE TITLE
Default Docker health check fails

### ISSUE THREAD

**Issue by lbogdan** [minio/minio#7623]
## Expected Behavior
Default Docker health check succeeds.

## Current Behavior
Default Docker health check fails.

## Steps to Reproduce (for bugs)
1. Start a `minio` container with no special configuration.
2. Wait a few minutes.
3. Run `docker inspect minio_container`:
```json
            "Health": {
                "Status": "unhealthy",
                "FailingStreak": 4,
                "Log": [
                    {
                        "Start": "2019-05-06T11:47:44.722634026Z",
                        "End": "2019-05-06T11:47:44.812943192Z",
                        "ExitCode": 1,
                        "Output": "2019/05/06 11:47:44 no minio process found\n"
                    },
```

## Debugging
After a quick look at `healthcheck.go` I saw it uses `netstat` to detect the port `minio` is listening on https://github.com/minio/minio/blob/efbc665ad366364923f8e1439a0fd9fcdc812de9/dockerscripts/healthcheck.go#L56
and running the same inside the container shows
```
# netstat -ntlp
Active Internet connections (only servers)
Proto Recv-Q Send-Q Local Address           Foreign Address         State       PID/Program name
tcp        0      0 127.0.0.11:39145        0.0.0.0:*               LISTEN      -
tcp        0      0 :::9000                 :::*                    LISTEN      -
```
As you can see, there's no pids / names in the `PID/Program name` column, most probably because of
```
# ls -al /proc/1/exe
ls: /proc/1/exe: cannot read link: Permission denied
lrwxrwxrwx    1 root     root             0 May  6 11:41 /proc/1/exe
```
and because `healthcheck` looks for `minio` at https://github.com/minio/minio/blob/efbc665ad366364923f8e1439a0fd9fcdc812de9/dockerscripts/healthcheck.go#L71
it can't find it and so it fails.

## Your Environment
* Version used (`minio version`): `2019-04-23T23:50:36Z / RELEASE.2019-04-23T23-50-36Z`
* Docker version: `18.06.1-ce`
* Linux distro: Ubuntu `18.04.2 LTS`
* Operating System and version (`uname -a`): `Linux dev 4.15.0-48-generic #51-Ubuntu SMP Wed Apr 3 08:28:49 UTC 2019 x86_64 x86_64 x86_64 GNU/Linux`

**Comment by nitisht** [comment]
thanks for reporting, we’ll take a look @lbogdan

**Comment by nitisht** [comment]
@lbogdan are you running custom MinIO image with non-root user?

**Comment by lbogdan** [comment]
No, it's `minio/minio:latest`:
```
# id
uid=0(root) gid=0(root)
```

**Comment by nitisht** [comment]
I just tested locally and I can see netstat reporting PID/Program name (while running as a root user)

```
7cc2cb6f7ece:~# netstat -ntlp
Active Internet connections (only servers)
Proto Recv-Q Send-Q Local Address           Foreign Address         State       PID/Program name    
tcp        0      0 0.0.0.0:9000            0.0.0.0:*               LISTEN      1/minio
```

I understand this will be a problem if `minio` is running as non root user. We'll need to fix that

**Comment by lbogdan** [comment]
It works if I `cap_add SYS_PTRACE`:
```
# ls -al /proc/1/exe
lrwxrwxrwx    1 root     root             0 May  8 06:25 /proc/1/exe -> /usr/bin/minio
# netstat -lntp
Active Internet connections (only servers)
Proto Recv-Q Send-Q Local Address           Foreign Address         State       PID/Program name
tcp        0      0 127.0.0.11:46123        0.0.0.0:*               LISTEN      -
tcp        0      0 :::9000                 :::*                    LISTEN      1/minio
```

**Comment by harshavardhana** [comment]
So you are using a different cap on your containers by default @lbogdan?

**Comment by lbogdan** [comment]
No, with everything default it doesn't work, if I add `SYS_PTRACE` it starts working.

**Comment by lbogdan** [comment]
If I'd have to guess, I'd say it has to do with AppArmor / other kernel security enforcement mechanism.

**Comment by harshavardhana** [comment]
May be we would have to validate if the recent docker changed anything, to drop privileges.

**Comment by nitisht** [comment]
Tested on latest Docker 

```
Client:
 Version:           18.09.6
 API version:       1.39
 Go version:        go1.10.8
 Git commit:        481bc77
 Built:             Sat May  4 02:35:57 2019
 OS/Arch:           linux/amd64
 Experimental:      false

Server: Docker Engine - Community
 Engine:
  Version:          18.09.6
  API version:      1.39 (minimum version 1.12)
  Go version:       go1.10.8
  Git commit:       481bc77
  Built:            Sat May  4 01:59:36 2019
  OS/Arch:          linux/amd64
  Experimental:     false
```

and `edge` MinIO Docker image, I am able to run netstat and get the MinIO program name

```
$ docker exec -it 9b81a03cd996 /bin/sh
/ # netstat -ntlp
Active Internet connections (only servers)
Proto Recv-Q Send-Q Local Address           Foreign Address         State       PID/Program name    
tcp        0      0 :::9000                 :::*   
```

Nevertheless, we're moving to a non-root user mechanism in MinIO image with https://github.com/minio/minio/pull/7569 . This PR removes the dependency on `Program name` output from netstat. So, once this PR is merged, your problem should be fixed too.

**Comment by harshavardhana** [comment]
This is fixed in #7569
